.github/workflows/check-merge-freeze.yml

```diff
@@ -0,0 +1,22 @@
+name: Merge freeze
+
+on:
+  pull_request:
+    types: [opened, ready_for_review, synchronize, reopened, labeled, unlabeled]
+    branches: [main]
+  merge_group:
+    types: [checks_requested]
+
+jobs:
+  check-merge-freeze:
+    name: Check
+    if: ${{ !contains(github.event.pull_request.labels.*.name, 'release:prepare') }}
```

Comment:
I wonder if we should call the label something that more explicitly relates to the freeze. Something like `release:halt-merges` or `release:freeze-main`?

Reply:
That's a good idea. Maybe `release:freeze` is clear enough?